lib/src/modules/task.cc

```diff
@@ -240,30 +239,49 @@ static std::string createUrlEndpoint(const lth_jc::JsonContainer& uri) {
 //    the same task file.
 // The downloaded task file's permissions will be set to rwx for user and rx for
 // group for non-Windows OSes.
-static void downloadTaskFile(const std::vector<std::string>& master_uris,
+static bool downloadTaskFile(const std::vector<std::string>& master_uris,
                              lth_curl::client& client,
                              const fs::path& file_path,
                              const lth_jc::JsonContainer& uri) {
-    auto url = master_uris[0] + createUrlEndpoint(uri);
-    lth_curl::request req(url);
-    // timeout from connection after one minute, can configure
-    req.connection_timeout(60000);
+    auto endpoint = createUrlEndpoint(uri);
+    for (auto& master_uri : master_uris) {
+        auto url = master_uri + endpoint;
+        lth_curl::request req(url);
+        // timeout from connection after one minute, can configure
+        req.connection_timeout(60000);
+        try {
 #ifndef _WIN32
-    client.download_file(req, file_path.string(), NIX_TASK_FILE_PERMS);
+            client.download_file(req, file_path.string(), NIX_TASK_FILE_PERMS);
 #else
-    client.download_file(req, file_path.string());
+            client.download_file(req, file_path.string());
 #endif
+        } catch (lth_curl::http_file_download_exception& e) {
```

Comment:
That's not true, this needs to exclude http_file_operation_exception.

Also, Leatherman 1.1.2 is tagged, so should be usable.

Reply:
Thanks for that catch, you're right! Making the fix.

Reply:
Ok, this should be addressed by Leatherman 1.1.3.